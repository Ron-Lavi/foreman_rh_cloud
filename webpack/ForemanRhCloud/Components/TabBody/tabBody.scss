.tab_body {
  margin-bottom: 0;
}
