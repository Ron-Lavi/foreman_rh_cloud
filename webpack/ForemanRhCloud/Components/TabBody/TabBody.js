import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'patternfly-react';
import Terminal from '../Terminal';
import ScheduledRun from '../ScheduledRun';
import './tabBody.scss';

const TabBody = ({ exitCode, logs, error, scheduled }) => (
  <Grid.Row className="tab_body">
    <Terminal logs={logs} exitCode={exitCode} error={error} />
    <ScheduledRun date={scheduled} />
  </Grid.Row>
);

TabBody.propTypes = {
  exitCode: PropTypes.string,
  logs: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.string),
    PropTypes.string,
  ]),
  error: PropTypes.string,
  scheduled: PropTypes.string,
};

TabBody.defaultProps = {
  exitCode: '',
  logs: null,
  error: null,
  scheduled: null,
};
export default TabBody;
