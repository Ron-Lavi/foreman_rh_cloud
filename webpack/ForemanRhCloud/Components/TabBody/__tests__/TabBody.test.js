import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import TabBody from '../TabBody';

const fixtures = {
  'render without Props': {},
  /** fixtures, props for the component */
};

describe('TabBody', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(TabBody, fixtures));
});
