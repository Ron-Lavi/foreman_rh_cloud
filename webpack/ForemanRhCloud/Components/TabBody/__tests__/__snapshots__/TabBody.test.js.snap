// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBody rendering render without Props 1`] = `
<Row
  bsClass="row"
  className="tab_body"
  componentClass="div"
>
  <Terminal
    autoScroll={true}
    error={null}
    exitCode=""
    logs={null}
  />
  <Connect(ScheduledRun)
    date={null}
  />
</Row>
`;
